src/fiber/fiber.ml

```diff
@@ -229,7 +229,9 @@ let sequential_iter l ~f =
     match l with
     | [] -> return ()
     | x :: l ->
+      EC.add_refs 1;
       let* () = f x in
+      EC.deref ();
```

Comment:
That doesn't seem right. `sequention_iter` is a function that only uses the public API, so it can be implemented outside of the `Fiber` module. As such, it shouldn't require the use of the private `add_refs` and `deref` functions.

Reply:
BTW, it's not due to `sequential_iter`, I added another test that reproduce the error with just `finalize`